AsyncAwait.Task2.CodeReviewChallenge/Middleware/StatisticMiddleware.cs

```diff
@@ -1,42 +0,0 @@
-﻿using System;
-using System.Threading;
-using System.Threading.Tasks;
-using AsyncAwait.Task2.CodeReviewChallenge.Headers;
-using CloudServices.Interfaces;
-using Microsoft.AspNetCore.Http;
-
-namespace AsyncAwait.Task2.CodeReviewChallenge.Middleware;
-
-public class StatisticMiddleware
-{
-    private readonly RequestDelegate _next;
-
-    private readonly IStatisticService _statisticService;
-
-    public StatisticMiddleware(RequestDelegate next, IStatisticService statisticService)
-    {
-        _next = next;
-        _statisticService = statisticService ?? throw new ArgumentNullException(nameof(statisticService));
-    }
-
-    public async Task InvokeAsync(HttpContext context)
-    {
-        string path = context.Request.Path;
-
-        var staticRegTask = Task.Run(
-            () => _statisticService.RegisterVisitAsync(path)
-                .ConfigureAwait(false)
-                .GetAwaiter().OnCompleted(UpdateHeaders));
-        Console.WriteLine(staticRegTask.Status); // just for debugging purposes
-
-        void UpdateHeaders()
-        {
-            context.Response.Headers.Add(
-                CustomHttpHeaders.TotalPageVisits,
-                _statisticService.GetVisitsCountAsync(path).GetAwaiter().GetResult().ToString());
-        }
-
-        Thread.Sleep(3000); // without this the statistic counter does not work
-        await _next(context);
-    }
```

Comment:
GetVisitsCountAsync(path).GetAwaiter().GetResult() is a blocking operation and a bad pratice. This code is refactored to two await expressions in the CloudService class GetVisitsCount method. This allows us to get rid of the Thread.Sleep statement.